modelon/impact/client/entities/project.py

```diff
@@ -419,18 +419,45 @@ def get_contents(self) -> List[ProjectContent]:
             for content in self._project_definition.content
         ]
 
-    def get_content_by_name(self, name: str) -> Optional[ProjectContent]:
+    def get_content_by_name(
+        self, name: str, content_type: Optional[ContentType] = None
+    ) -> Optional[ProjectContent]:
         """Gets the first matching project content with the given name.
 
+        Parameters:
+
+            name --
+                The name of the content.
+
+            content_type --
+                The type of the project content.
+        Example::
+            from modelon.impact.client import ContentType
+            project.get_content_by_name(name, ContentType.MODELICA)
+        """
+        contents = self.get_contents()
+        if content_type:
+            contents = [
+                content for content in contents if content.content_type == content_type
+            ]
```

Comment:
supper minor but you can use `c` and have it as a generator, will be more symmetrical to the code bellow
`contents = (c for c in contents if c.content_type == content_type)`